src/parser.cpp

```diff
@@ -219,65 +209,63 @@ namespace ada::parser {
                 url.is_valid = false;
                 return url;
               }
-              // Decrease pointer by the number of code points in buffer plus one,
-              // set buffer to the empty string, and set state to host state.
+              // Decrease pointer by the number of code points in buffer plus one and set state to host state.
               input_position -= authority_view.length() + 1;
               state = ada::state::HOST;
+              input_position++;
```

Comment:
Can't we combine this with the previous line and do

```
input_position -= authority_view.length();
```

because otherwise you are decrementing by one only to immediately increment by one.

Reply:
Yes you're right.